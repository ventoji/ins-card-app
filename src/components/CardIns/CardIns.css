
.ins-card{
    height: 27rem;
    display: flex;
    flex-direction: column;
    box-shadow: 0 1.1rem 1.1rem rgba(0, 0, 0, 0.41);
    margin: 2rem;
    width: 27rem;
}

.ins-card__image-container{
    height: 15rem;
}

.ins-card__image{
    width: 100%;
    height: 100%;
    opacity: 0.8;
}

.ins-card__description{
    font-size: 1.3rem;
    padding: 2rem;
    background: #fff;
}

.ins-card__description p{
    word-break: break-all;
}

.ins-card__title{
    z-index: 10;
    position: absolute;
    color: #fff;
    font-size: 2.3rem;
    padding: 1rem;
 /*   top: 12.8rem; */

}