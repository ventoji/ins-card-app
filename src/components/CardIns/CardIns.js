
import React from 'react';
import PropTypes from 'prop-types';
import './CardIns.css';

/** Component for a unique card */
const CardIns = ({
    title,
    description,
    imageUrl
}) => {
    return (
        <div className="ins-card">

            <div className="ins-card__image-container">
            <h5 className="ins-card__title">{title}</h5>
               <img 
                className="ins-card__image"
                src={imageUrl} 
                alt="card pic"
                />
            </div> 
            
            <div className="ins-card__description">
                <p>{description}</p>
            </div>          
            
        </div>
    );
};


CardIns.defaultProps = {
    title: 'Moto GP',
    description: 'Some description for the card, very nice',
    imageUrl: 'https://images.freeimages.com/images/large-previews/adf/sun-burst-1478549.jpg'
};

CardIns.propTypes = {
    /**
     *  Image for the card provided by a url
     */
    imageUrl: PropTypes.string,

    /**
     *  Description for the card
     */
    description: PropTypes.string.isRequired,

    /**
     *  Title for the card 
     */
    title: PropTypes.string.isRequired
}


export default CardIns;