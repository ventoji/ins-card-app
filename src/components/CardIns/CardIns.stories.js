import React from "react";
import { Provider } from 'react-redux';
import {CardIns} from "./CardIns";
import { text } from "@storybook/addon-knobs";
import configureStore from '../../store/configureStore';
const store = configureStore();

export default {
  title: "Components/CarIns",
  component: CardIns,
  decorators: [
    (Story) => (
      <Provider store={store}>
        <Story />
      </Provider>
    ),
    //   withReduxDecorator
  ],
};

const Template = (args) => (
  <div className="ins-row-stories">
    {" "}
    <CardIns {...args} />{" "}
  </div>
);

export const DefaultCard = Template.bind({});
DefaultCard.args = {
  title: text("label", "Moto GP"),
  description:
    "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus e",
  imageUrl:
    "https://images.freeimages.com/images/large-previews/adf/sun-burst-1478549.jpg",
};
