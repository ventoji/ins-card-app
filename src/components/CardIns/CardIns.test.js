import React from "react";
import { render } from "@testing-library/react";

import {CardIns} from "./CardIns";

const CARD_DETAILS = {
  title: "Moto GP",
  description: "Some description for the card, very nice",
  imageUrl:
    "https://images.freeimages.com/images/large-previews/adf/sun-burst-1478549.jpg",
};

describe("Card component", () => {
  it("should be defined", () => {
    const { container } = render(<CardIns />);
    expect(container).toBeInTheDocument();
    //    expect(container.querySelectorAll('img')).toHaveLength(1);
    expect(container.querySelectorAll("div")[0]).toHaveClass("ins-card");
    expect(container.querySelectorAll("div")[1]).toHaveClass(
      "ins-card__image-container"
    );
    expect(container.querySelectorAll("div")[3]).toHaveClass(
      "ins-card__description"
    );
  });

  it("should render card with defaults", () => {
    const { getByText } = render(<CardIns />);
    let element;

    element = getByText("Moto GP");
    expect(element).toBeInTheDocument();
    // console.log(element.nodeName);
    expect(element.nodeName).toBe("H5");

    element = getByText("Some description for the card, very nice");
    expect(element).toBeInTheDocument();
  });

  it("should render card with provided details", () => {
    const { getByText } = render(<CardIns {...CARD_DETAILS} />);
    let element = getByText("Moto GP");
    expect(element).toBeInTheDocument();
    //    expect(container.querySelector('img').src).toBe(CARD_DETAILS.imageUrl);
  });
});
