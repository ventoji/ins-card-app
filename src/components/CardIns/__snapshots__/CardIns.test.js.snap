// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render with snapshot 1`] = `
<div
  className="ins-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="ins-card__image-container"
  >
    <div
      className="sc-bdvvaa hTVhwk"
    >
      <h5
        className="ins-card__title"
      >
        Moto GP
      </h5>
    </div>
  </div>
  <div
    className="ins-card__description"
  >
    <div
      className="ins-card__controls "
    >
      <div
        className="ins-form__group"
      >
        <button
          className="ins-button ins-button--small ins-button--primary"
          onClick={[Function]}
          type="button"
        >
          edit
        </button>
      </div>
      <div
        className="ins-form__group"
      >
        <button
          className="ins-button ins-button--small ins-button--danger"
          onClick={[Function]}
          type="submit"
        >
          delete
        </button>
      </div>
    </div>
    <p>
      Some description for the card, very nice
    </p>
  </div>
</div>
`;
